pkg/polaris/config.go

```diff
@@ -268,144 +224,217 @@ func UserAccountFromFile(file, name string, allowEnvOverride bool) (*UserAccount
+func userAccountFromFile(file, name string) (UserAccount, error) {
+	expFile, err := expandPath(file)
 	if err != nil {
-		return nil, err
+		return UserAccount{}, fmt.Errorf("failed to expand file path: %s", err)
 	}
 
-	// Validate.
-	if account.Name == "" {
-		return nil, errors.New("invalid service account name")
-	}
-	if account.ClientID == "" {
-		return nil, errors.New("invalid service account client id")
+	buf, err := os.ReadFile(expFile)
+	if err != nil {
+		return UserAccount{}, fmt.Errorf("failed to read user account file: %s", err)
 	}
-	if account.ClientSecret == "" {
-		return nil, errors.New("invalid service account client secret")
+
+	var accounts map[string]UserAccount
+	if err := json.Unmarshal(buf, &accounts); err != nil {
+		return UserAccount{}, fmt.Errorf("failed to unmarshal user account file: %s", err)
 	}
-	if account.AccessTokenURI == "" {
-		return nil, errors.New("invalid service account access token uri")
+
+	account, ok := accounts[name]
+	if !ok {
+		return UserAccount{}, fmt.Errorf("failed to lookup user account %q: %w", name, graphql.ErrNotFound)
 	}
+	account.Name = name
 
-	return &account, nil
+	return account, nil
 }
 
-// serviceAccountFromFile returns a ServiceAccount from the specified RSC
-// service account file.
-func serviceAccountFromFile(file string) (ServiceAccount, error) {
-	expFile, err := expandPath(file)
+// initUserAccount validates the user account data and initializes the
+// additional fields.
+func initUserAccount(account *UserAccount) error {
+	if account.Name == "" {
+		return errors.New("invalid user account name")
+	}
+	if account.Username == "" {
+		return errors.New("invalid user account username")
+	}
+	if account.Password == "" {
+		return errors.New("invalid user account password")
+	}
+	if account.URL == "" {
+		account.URL = fmt.Sprintf("https://%s.my.rubrik.com/api", account.Name)
+	}
+
+	// Derive fields.
+	u, err := url.ParseRequestURI(account.URL)
 	if err != nil {
-		return ServiceAccount{}, fmt.Errorf("failed to expand file path: %s", err)
+		return fmt.Errorf("invalid url: %s", err)
+	}
+	fqdn := u.Hostname()
+	i := strings.Index(fqdn, ".")
+	if i == -1 {
+		return errors.New("invalid url: no account name found")
 	}
+	account.accountName = fqdn[:i]
+	account.accountFQDN = fqdn
+	account.apiURL = account.URL
+	account.tokenURL = account.URL + "/session"
 
-	buf, err := os.ReadFile(expFile)
+	return nil
+}
+
+// ServiceAccount holds an RSC ServiceAccount configuration. The Name field
+// holds the name of the service account and not the name of the RSC account.
+type ServiceAccount struct {
+	ClientID       string `json:"client_id"`        // Client ID.
+	ClientSecret   string `json:"client_secret"`    // Client secret.
+	Name           string `json:"name"`             // Service account name.
+	AccessTokenURI string `json:"access_token_uri"` // Access token URI.
+
+	accountName string
+	accountFQDN string
+	apiURL      string
+	envOverride bool
+	tokenURL    string
```

Comment:
nit: comments for additional struct values

Reply:
We primarily document the public API, in addition to this the `ServiceAccount`/`UserAccount` struct has additional documentation for the public access functions for each of the fields.